runtime/ftplugin/tera.vim

```diff
@@ -1,13 +1,29 @@
 " Vim filetype plugin file
 " Language:             Tera
 " Maintainer:           Muntasir Mahmud <muntasir.joypurhat@gmail.com>
-" Last Change:          2025 Mar 06
+" Last Change:          2025 Mar 08
 
 if exists("b:did_ftplugin")
   finish
 endif
 let b:did_ftplugin = 1
 
+setlocal autoindent
+
 setlocal commentstring={#\ %s\ #}
+setlocal comments=s:{#,e:#}
+
+if exists("loaded_matchit")
+  let b:match_ignorecase = 0
+  let b:match_words = '{#:##\|#},{% *if:{% *else\>:{% *elif\>:{% *endif %},{% *for\>:{% *endfor %},{% *macro\>:{% *endmacro %},{% *block\>:{% *endblock %},{% *filter\>:{% *endfilter %},{% *set\>:{% *endset %},{% *raw\>:{% *endraw %},{% *with\>:{% *endwith %}'
+endif
+
+setlocal includeexpr=substitute(v:fname,'\\([^.]*\\)$','\\1','g')
+setlocal suffixesadd=.tera
+
+setlocal expandtab
+setlocal shiftwidth=2
+setlocal softtabstop=2
 
-let b:undo_ftplugin = "setlocal commentstring<"
+let b:undo_ftplugin = "setlocal autoindent< commentstring< comments< " .
+      \ "include< includeexpr< suffixesadd< expandtab< shiftwidth< softtabstop< formatoptions<"
```

Comment:
The include and formatoptions have not been set, so don't need to be reset.

But you can could also add: 
```vim
let b:undo_ftplugin .= "|unlet! b:match_ignorecase b:match_words"
```